boto3/resources/model.py

```diff
@@ -312,6 +312,66 @@ def batch_actions(self):
 
         return actions
 
+    def _get_has_definition(self):
+        """
+        Get a ``has`` relationship definition from a model, where the
+        service resource model is treated special in that it contains
+        a relationship to every resource defined for the service. This
+        allows things like ``s3.Object('bucket-name', 'key')`` to
+        work even though the JSON doesn't define it explicitly.
+
+        @rtype: dict
+        @return: Mapping of names to subresource and reference
+                 definitions.
+        """
+        if self.name not in self._resource_defs:
+            # This is the service resource, so let us expose all of
+            # the defined resources as subresources.
+            definition = {}
+
+            for name, resource_def in self._resource_defs.items():
+                # It's possible for the service to have renamed a
+                # resource, so we need to take that into account and
+                # search for the same resource type.
+                has_items = self._definition.get('has', {}).items()
+                for has_name, has_def in has_items:
+                    if has_def.get('resource', {}).get('type') == name:
+                        definition[has_name] = has_def
+                        break
+                else:
```

Comment:
It's from the `for` loop. If it never breaks, then the `else` is executed. I've since removed it to simplify the code and fix a potential case I hadn't considered so it should now be easier to understand.